tsconfig.types.json

```diff
@@ -7,17 +7,11 @@
     "declaration": true,
     "emitDeclarationOnly": true,
     "declarationMap": true,
+    "rootDir": "./src"
   },
   "include": [
     "src/core/server/index.ts",
     "src/core/public/index.ts",
-    "src/plugins/data/server/index.ts",
```

Comment:
After moving the code from `data_enhanced` to `data` this type generation for docs stopped working because `data` now references code from x-pack (task_manager, security) and this changed the output folder structure of this types. 

Since we don't use these types for app services plugins generation anymore, I just removed them. I added the `rootDir` to preserve the previous file tree output structure.